src/server/active_tran_server.cpp

```diff
@@ -351,23 +369,34 @@ void active_tran_server::receive_data_page (cubpacking::unpacker &upk)
   std::string message;
   upk.unpack_string (message);
 
-  int error_code;
-  std::memcpy (&error_code, message.c_str (), sizeof (error_code));
+  if (message.size () == sizeof (int))
+    {
+      // We have an error.
+      int error_code;
+      std::memcpy (&error_code, message.c_str (), sizeof (error_code));
 
-  if (prm_get_bool_value (PRM_ID_ER_LOG_READ_DATA_PAGE))
+      if (prm_get_bool_value (PRM_ID_ER_LOG_READ_DATA_PAGE))
+	{
+	  _er_log_debug (ARG_FILE_LINE, "Received data page message from Page Server. Error: %d \n", error_code);
+	}
+    }
+  else
     {
-      if (error_code == NO_ERROR)
+      // We have a page.
```

Comment:
Add safe-guard that the message size matches the log page size.
```suggestion
      assert (message.size () == db_io_page_size ());

      // We have a page.
```